src/providers/EtherspotContextProvider.tsx

```diff
@@ -540,22 +533,20 @@ const EtherspotContextProvider = ({
     if (onLogout) onLogout();
   }, [setProvider, setProviderAddress, setAccountAddress, onLogout]);
 
-  useEffect(() => {
-    const handleBalanceGet = async () => {
+  const handleGetBalances = useCallback(
+    async (force?: boolean) => {
       if (!sdk || !accountAddress) return;
-      await getSmartWalletBalancesByChain(accountAddress, supportedChains);
-    };
-    handleBalanceGet();
-  }, [supportedChains, sdk, accountAddress]);
 
-  useEffect(() => {
```

Comment:
Hey man! Great work here! Just a suggestion about this useEffect call. I feel it can be optimised by
1) Using Promise.all
2) Removing unnecessary conditions from the useEffect dependencies array to avoid triggering unnecessary function calls.
Something like this:
```
useEffect(async () => {
  if (!!force || !smartWalletBalanceByChain.length) {
    await Promise.all([
      getSmartWalletBalancesByChain(accountAddress, supportedChains),
      getKeybasedWalletBalancesPerChain(providerAddress, supportedChains),
    ]);
  }
}, [force, accountAddress, providerAddress]);
```
But overall it looks good to me.